a10_octavia/common/utils.py

```diff
@@ -40,13 +41,28 @@ def validate_params(rack_info):
                                                   'username', 'password', 'device_name')):
             if validate_ipv4(rack_info['ip_address']):
                 return True
-            LOG.error('Invalid IP address given ' + rack_info['ip_address'])
-    LOG.error('Configuration of `devices` under [RACK_VTHUNDER] is invalid. '
-              'Please check your configuration. The params `project_id`, '
-              '`ip_address`, `username`, `password` and `device_name` cannot be None ')
+            raise exceptions.ValidationException(detail=_(
+                'Invalid IP address given ' + rack_info['ip_address']))
+    raise exceptions.ValidationException(detail=_(
+        'Configuration of `devices` under [RACK_VTHUNDER] is invalid. '
+        'Please check your configuration. The params `project_id`, '
+        '`ip_address`, `username`, `password` and `device_name` cannot be None '))
     return False
 
 
+def check_partition_collision(rack_device):
```

Comment:
Since we are calling `validate_params()` before calling this `check_partition_collision()` method, we are making sure - `ip_address` is required along with `project_id`, `username`, `password` and `device_name`. If any one among this is not provided, we are raising exception, with appropriate message.

Reply:
Ok so you want me to store all the duplicated (wrong configs) in a set and then listing them out in the error message.  Modifying the logic accordingly.